distributed/tests/test_client.py

```diff
@@ -1684,6 +1684,21 @@ def test_upload_file_exception_sync(c):
             c.upload_file(fn)
 
 
+@gen_cluster(client=True, nthreads=[])
+async def test_upload_file_new_worker(c, s):
+    def g():
+        import myfile
+
+        return myfile.x
+
+    with tmp_text("myfile.py", "x = 123") as fn:
+        await c.upload_file(fn)
+        await Worker(s.address, loop=s.loop)
+        x = await c.submit(g)
+
+        assert x == 123
```

Comment:
I suspect that tests will yell at you for leaving a lingering Worker .  We might want to try `async with Worker(...) as w` instead.  Also, I think that the `loop=` keyword is probably no longer necessary.